src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -9,20 +9,21 @@ package internal
 
 import scala.collection.immutable
 import scala.collection.mutable.ListBuffer
-import util.{ Statistics, shortClassOfInstance, StatisticsStatics }
+import util.{Parallel, Statistics, StatisticsStatics, shortClassOfInstance}
 import Flags._
 import scala.annotation.tailrec
-import scala.reflect.io.{ AbstractFile, NoAbstractFile }
+import scala.reflect.io.{AbstractFile, NoAbstractFile}
 import Variance._
 
 trait Symbols extends api.Symbols { self: SymbolTable =>
   import definitions._
   import statistics._
 
-  protected var ids = 0
-  def getCurrentSymbolIdCount: Int = ids
+  private[this] var ids = 0
+  def getCurrentSymbolIdCount: Int = Parallel.synchronizeAccess(IdsLock)(ids)
+  private[this] object IdsLock
```

Comment:
Maybe we can keep all locks in the `SymbolTable`?
We know it have proper lifecycle/scope if there is more than one mirror involved.
Also, seeing them all in one place might create a little bit of clutter but will also allow us to see all of them at once and possibly reason better about them (like for example: "Oh, we have lock on Ids and Symbols, but should they be independent locks? Nope, they need to be synchronized together!" - just an example, I'm not trying to say that it's the case here).

Reply:
This one is different and IMHO requires its own lock. Will add comment in PR